MultiServer.py

```diff
@@ -2536,6 +2536,25 @@ async def main(args: argparse.Namespace):
             logging.info("No file selected. Exiting.")
             import sys
             sys.exit(1)
+        elif not args.disable_save:
+            import tempfile
+            import os
+            import sys
+            try:
+                common = os.path.commonpath((tempfile.gettempdir(), data_filename))
+                if not os.path.samefile(tempfile.gettempdir(), common):
+                    raise ValueError
+
+            except ValueError:
```

Comment:
This flow seems very odd. I would at least leave a comment why it uses ValueError in the happy path